File: Day04/src/Day04/Sources/ObjCModule/Day04_02.m
//
//  Day04_02.m
// 
//
//  Created by Artour Ilyasov on 17.03.2023.
//

#import <Foundation/Foundation.h>
#import "Day04_02.h"

@implementation Barista

- (instancetype)init {
    self = [super init];
    if (self) {
        _name = @" ";
        _surname = @" ";
    }
    return self;
}

- (void)brew:(Coffee*)coffee {
    printf("Proccesing brewing coffee...\n");
    printf("Your %s is ready!\n", [coffee.name UTF8String]);
}

@end

@implementation Coffee

- (instancetype)init {
    self = [super init];
    if (self) {
        _name = @" ";
        _price = 0;
    }
    return self;
}

@end

File: Day04/src/Day04/Sources/Day04_02/Day04_02.swift
//
//  File.swift
//  
//
//  Created by Artour Ilyasov on 17.03.2023.
//

import Foundation
import ObjCModule

@main
public struct Day04_02 {
    public static func main() {
        print("""
              Choose barista:\n
              1. Man\n
              2. Machine\n
              """)
        
        
    }
}


class BaristaMachine {
    let modelName: String = ""
    
    func brew() {
        print("Coffee is brewing. 1 minute - time left")
        print("Coffee is ready!")
    }
    
    
    init(coffee: Coffee) {}
}
